packages/socket/patches/socket.io-parser+4.0.4.patch

```diff
@@ -146,7 +146,11 @@ index 4b7c234..95469f7 100644
 +    known.push(obj)
      if (Array.isArray(obj)) {
          for (let i = 0, l = obj.length; i < l; i++) {
-             if (hasBinary(obj[i])) {
+-            if (hasBinary(obj[i])) {
++            if (hasBinary(obj[i], known)) {
+                 return true;
+             }
+         }
```

Comment:
The cause of the problem was the incorrect recursive call of `hasBinary` inside check Array section. Without `known` argument being passed, when checking arrays that have circular references, `hasBinary` always starts from fresh.

Reply:
:facepalm: x 10000000